File-Formats/DocIO/Working-with-Find-and-Replace.md

```diff
@@ -799,6 +799,116 @@ Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Replace.docx", "applic
+        //Save a Word document to MemoryStream.
+        MemoryStream stream = new MemoryStream();
+        await document.SaveAsync(stream, FormatType.Docx);
+        //Save the stream as a Word document file in the local machine.
+        Save(stream, "Sample.docx");
+	//Please refer the below link to save Word document in UWP platform
+	//https://help.syncfusion.com/file-formats/docio/create-word-document-in-uwp#save-word-document-in-uwp
+    }
+}
+{% endhighlight %}
+
+{% highlight c# tabtitle="ASP.NET Core" %}
+//Open the file as Stream.
+using (FileStream docStream = new FileStream("Input.docx", FileMode.Open, FileAccess.Read))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        //Access the specific paragraph in the Word document.
+        TextBodyItem textBodyItem = document.Sections[0].Paragraphs[3] as WParagraph;
+        //Find the next occurrence of specified text from the previous paragraph.
+        TextSelection textSelections = document.FindNext(textBodyItem, new Regex("Adventure Works Cycles"));
+        //Replace the text extended to two paragraphs with the above selected text.
+        document.ReplaceSingleLine("CompanyName", textSelections, true, true);
+        //Save the Word document to the MemoryStream.
+        MemoryStream outputStream = new MemoryStream();
+        document.Save(outputStream, FormatType.Docx);
+        outputStream.Position = 0;
+        //Download as Word document in the browser.
+        return File(outputStream, "application/msword", "Sample.docx");
+    }
+}
+{% endhighlight %}
+
+{% highlight c# tabtitle="Xamarin" %}
+//Open the file as Stream.
+using (Stream docStream = typeof(App).GetTypeInfo().Assembly.GetManifestResourceStream("Sample.Assets.Input.docx"))
+{
+    //Load the file stream into a Word document.
+    using (WordDocument document = new WordDocument(docStream, FormatType.Docx))
+    {
+        //Access the specific paragraph in the Word document.
+        TextBodyItem textBodyItem = document.Sections[0].Paragraphs[3] as WParagraph;
+        //Find the next occurrence of specified text from the previous paragraph.
+        TextSelection textSelections = document.FindNext(textBodyItem, new Regex("Adventure Works Cycles"));
+        //Replace the text extended to two paragraphs with the above selected text.
+        document.ReplaceSingleLine("CompanyName", textSelections, true, true);
+        //Save the Word document to the MemoryStream.
+        MemoryStream outputStream = new MemoryStream();
+        document.Save(outputStream, FormatType.Docx);
+        //Save the stream as a file in the device and invoke it for viewing. 
+        Xamarin.Forms.DependencyService.Get<ISave>().SaveAndView("Sample.docx", "application/msword", outputStream);
+	//Please download the helper files from the below link to save the stream as file and open the file for viewing in Xamarin platform
```

Comment:
//Please download the helper files from the following link to save the stream as a file and open the file for viewing in the Xamarin platform.